modules/transactions.js

```diff
@@ -73,39 +73,90 @@ __private.attachApi = function () {
 
 __private.list = function (filter, cb) {
 	var sortFields = sql.sortFields;
-	var params = {}, where = [], owner = '';
+	var params = {},
+		where = [],
+		err,
+		owner = '',
+		allowedFieldsMap = {
```

Comment:
It is just an idea. Maybe provide also some level regarding field types? After that, you are using them grouped into categories like 
- 0 is allowed for few `['fromTimestamp', 'minAmount', 'minConfirmations', 'type', 'offset']` 
- array-like parameters`['senderIds', 'recipientIds', 'senderPublicKeys', 'recipientPublicKeys']`
- timestamps `['fromTimestamp', 'toTimestamp']`